float64  current_u0
